src/components/Discover/DiscoverSearch.tsx

```diff
@@ -223,29 +220,10 @@ export default function DiscoverSearch() {
   );
 
   useEffect(() => {
-    if (searchQueryForSearch && !isLoading) {
-      if (lastSearchQuery !== searchQueryForSearch) {
```

Comment:
previous logic had slightly funky behavior that would cause issues where the "old results" might still be saved (for instance if the query changed while the results were still loading, it would exit after the first check on line 226 but after the results would load, it would then exit at line 227 thinking that the query hasn't changed by then). I also simplified the dependencies to strictly what is needed, and relied on the debounced `searchQueryForSearch` instead of both that and the original `searchQuery`